Das 2284

sudha-murthy opened the pull request:
## Description

- The updates provides a solution to a test case where the granule has no dimensions and multiple coordinate variables are in the same group. The fix calculates the cached internal dimension names created from the coordinates to include the full name path of both the latitude and longitude coordinate variables. 
- Also added the bug fix in get_valid_sample_pts for 3D variables.
- The issue encountered in DAS-2317 is corrected with configuration for coordinate variables with no coordinate attributes in SMAP L3

## Jira Issue ID
DAS-2284

## Local Test Steps
- The unit tests should pass. 
- Hoss  SMAP L3 end to end requests should pass
More Test instructions on the ticket DAS-2284

## PR Acceptance Checklist
* [X ] Jira ticket acceptance criteria met.
* [X ] `CHANGELOG.md` updated to include high level summary of PR changes.
* [ X] `docker/service_version.txt` updated if publishing a release.
* [X ] Tests added/updated and passing.
* [ ] Documentation updated (if needed).

sudha-murthy commented on hoss/coordinate_utilities.py:
```diff
@@ -110,38 +110,33 @@ def get_dimension_array_names(
-            varinfo, latitude_coordinates[0]
-        )
-
-    # Given variable variable has no coordinate attribute itself,
```
latitude and longitude coordinates are required in this solution. The hoss_config.json has been updated for coordinates which do not have coordinate attributes

sudha-murthy commented on tests/data/SC_SPL3SMP_008_fake.dmr:
```diff
@@ -111,6 +111,90 @@
                 <Value>Y</Value>
             </Attribute>
         </Float64>
+        <Float32 name="same_folder_longitude1">
```
provides a test case for testing tow coordinate sets in the same group

sudha-murthy commented on hoss/hoss_config.json:
```diff
@@ -400,6 +400,132 @@
       ],
       "_Description": "Ensure variables in /Soil_Moisture_Retrieval_Data_Polar_PM group point to correct coordinate variables."
     },
+    {
```
Coordinate attributes added to coordinate variables that do not have them

joeyschultz commented on hoss/coordinate_utilities.py:
```diff
@@ -110,38 +110,33 @@ def get_dimension_array_names(
     # to define variable spatial dimensions.
```
This comment should be updated (specifically line 109 which I am not able to add this comment to) now that both the latitude and longitude are used to define the dimension_names.

sudha-murthy replied:
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/31/commits/38f6b50bf01336a986d8edfb4658288de9e07437

joeyschultz reviewed (changes requested):
Overall, the changes look good! I just had a few minor comments. The unit tests all passed, but I still need to go through the rest of the test instructions.

sudha-murthy commented on hoss/hoss_config.json:
```diff
@@ -504,6 +630,24 @@
+          "Value": "land cover type"
+        }
+      ],
+      "_Description": "The pseudo-dimension variable is here supplemented with variable attributes (as if it was a dimension variables) to clarify the dimension name"
```
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/31/commits/38f6b50bf01336a986d8edfb4658288de9e07437

sudha-murthy commented on hoss/hoss_config.json:
```diff
@@ -400,6 +400,132 @@
+      "Attributes": [
+        {
+          "Name": "coordinates",
+          "Value": "/Soil_Moisture_Retrieval_Data_Polar/latitude, /Soil_Moisture_Retrieval_Data_Polar/longitude"
```
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/31/commits/38f6b50bf01336a986d8edfb4658288de9e07437

sudha-murthy commented on hoss/hoss_config.json:
```diff
@@ -400,6 +400,132 @@
+          "Value": "/Soil_Moisture_Retrieval_Data_Polar_AM/latitude, /Soil_Moisture_Retrieval_Data_Polar_AM/longitude"
+        }
+      ],
+      "_Description": "Ensure coordinate variables in /Soil_Moisture_Retrieval_Polar_Data_AM group have coordinate attributes."
```
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/31/commits/38f6b50bf01336a986d8edfb4658288de9e07437

sudha-murthy commented:
@joeyschultz @D-Auty - I have addressed the comments in the commit https://github.com/nasa/harmony-opendap-subsetter/pull/31/commits/38f6b50bf01336a986d8edfb4658288de9e07437
Please let me know if anything else needs to be addressed.

joeyschultz reviewed (approved):
Thanks for the updates Sudha!
